public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx

```diff
@@ -1,30 +1,118 @@
+export const GeneralSettings: React.FC<Props> = ({ dashboard }) => {
+  const [renderCounter, setRenderCounter] = useState(0);
 
-    const template = '<dashboard-settings dashboard="dashboard" />';
-    const scopeProps = { dashboard: this.props.dashboard };
-    this.angularCmp = loader.load(this.element, scopeProps, template);
-  }
+  const onFolderChange = (folder: { id: number; title: string }) => {
+    dashboard.meta.folderId = folder.id;
+    dashboard.meta.folderTitle = folder.title;
+    dashboard.meta.hasUnsavedFolderChange = true;
+  };
 
-  componentWillUnmount() {
-    if (this.angularCmp) {
-      this.angularCmp.destroy();
-    }
-  }
+  const onBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    dashboard[event.currentTarget.name as 'title' | 'description'] = event.currentTarget.value;
+  };
 
-  render() {
-    return <div ref={(ref) => (this.element = ref)} />;
-  }
-}
+  const onTooltipChange = (graphTooltip: SelectableValue<number>) => {
+    dashboard.graphTooltip = graphTooltip.value;
+    setRenderCounter(renderCounter + 1);
+  };
+
+  const onRefreshIntervalChange = (intervals: string[]) => {
+    dashboard.timepicker.refresh_intervals = intervals.filter((i) => i.trim() !== '');
+  };
+
+  const onNowDelayChange = (nowDelay: string) => {
+    dashboard.timepicker.nowDelay = nowDelay;
+  };
+
+  const onHideTimePickerChange = (hide: boolean) => {
+    dashboard.timepicker.hidden = hide;
+    setRenderCounter(renderCounter + 1);
+  };
+
+  const onTimeZoneChange = (timeZone: TimeZone) => {
+    dashboard.timezone = timeZone;
+    setRenderCounter(renderCounter + 1);
+  };
+
+  const onTagsChange = (tags: string[]) => {
+    dashboard.tags = tags;
+  };
+
+  const onEditableChange = (ev: React.FormEvent<HTMLInputElement>) => {
+    dashboard.editable = ev.currentTarget.checked;
+    setRenderCounter(renderCounter + 1);
+  };
+
+  return (
+    <>
+      <h3 className="dashboard-settings__header" aria-label={selectors.pages.Dashboard.Settings.General.title}>
+        General
+      </h3>
+      <div className="gf-form-group">
+        <InlineField label="Name" labelWidth={14}>
+          <Input name="title" onBlur={onBlur} defaultValue={dashboard.title} width={60} />
+        </InlineField>
+        <InlineField label="Description" labelWidth={14}>
+          <Input name="title" onBlur={onBlur} defaultValue={dashboard.description} width={60} />
+        </InlineField>
+        <InlineField label="Tags" labelWidth={14}>
+          <TagsInput tags={dashboard.tags} onChange={onTagsChange} />
+        </InlineField>
+        <FolderPicker
+          initialTitle={dashboard.meta.folderTitle}
+          initialFolderId={dashboard.meta.folderId}
+          onChange={onFolderChange}
+          enableCreateNew={true}
+          dashboardId={dashboard.id}
+        />
+        <InlineField label="Editable" labelWidth={14}>
+          <Switch value={dashboard.editable} onChange={onEditableChange} />
```

Comment:
I think this should use `InlineSwitch`, otherwise the alignment is off:
![Screenshot 2021-02-06 at 12 38 31](https://user-images.githubusercontent.com/8878045/107115877-39cc8100-6878-11eb-9fac-77939de03496.png)

The same goes for `Hide time picker`.